lib/spack/spack/package.py

```diff
@@ -2111,6 +2115,12 @@ def _run_default_build_time_test_callbacks(self):
             except AttributeError:
                 msg = 'RUN-TESTS: method not implemented [{0}]'
                 tty.warn(msg.format(name))
+            except ProcessError:
+                if self.ignore_test_failures:
+                    msg = 'RUN-TESTS: test failures detected [{0}]'
+                    tty.warn(msg.format(name))
```

Comment:
I would prefer to have this warning printed to the console rather than to spack_build_out.txt, but I didn't know how to do that.